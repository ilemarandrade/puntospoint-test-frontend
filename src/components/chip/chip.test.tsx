import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Chip from './';
import { CarCrashRounded } from '@mui/icons-material';

describe('Chip', () => {
  test('renders with label', () => {
    const { container } = render(<Chip label="Test Label" />);
    expect(screen.getByText('Test Label')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test('renders with icon', () => {
    const { container } = render(
      <Chip label="Test Label" icon={<CarCrashRounded />} />
    );
    expect(screen.getByText('Test Label')).toBeInTheDocument();
    expect(screen.getByTestId('CarCrashRoundedIcon')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test('renders with delete icon and calls onDelete', () => {
    const handleDelete = jest.fn();

    const { container } = render(
      <Chip label="Test Label" onDelete={handleDelete} />
    );

    const clearIcon = screen.getByTestId('ClearRoundedIcon');

    expect(clearIcon).toBeInTheDocument();

    fireEvent.click(clearIcon);
    expect(handleDelete).toHaveBeenCalledTimes(1);

    expect(container).toMatchSnapshot();
  });

  test('renders and calls onClick', () => {
    const handleClick = jest.fn();

    const { container } = render(
      <Chip label="Test Label" onClick={handleClick} />
    );

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();

    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);

    expect(container).toMatchSnapshot();
  });

  test('renders outlined chip', () => {
    const { container } = render(
      <Chip label="Test Label" variant="outlined" />
    );

    expect(screen.getByText('Test Label')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test('renders outlined chip with icon', () => {
    const { container } = render(
      <Chip label="Test Label" icon={<CarCrashRounded />} variant="outlined" />
    );
    expect(screen.getByText('Test Label')).toBeInTheDocument();
    expect(screen.getByTestId('CarCrashRoundedIcon')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test('renders outlined chip with delete', () => {
    const handleDelete = jest.fn();

    const { container } = render(
      <Chip label="Test Label" onDelete={handleDelete} variant="outlined" />
    );

    const clearIcon = screen.getByTestId('ClearRoundedIcon');

    expect(clearIcon).toBeInTheDocument();

    fireEvent.click(clearIcon);
    expect(handleDelete).toHaveBeenCalledTimes(1);

    expect(container).toMatchSnapshot();
  });
});

test('renders outlined chip and calls onClick', () => {
  const handleClick = jest.fn();

  const { container } = render(
    <Chip label="Test Label" onClick={handleClick} variant="outlined" />
  );

  const button = screen.getByRole('button');

  expect(button).toBeInTheDocument();

  fireEvent.click(button);
  expect(handleClick).toHaveBeenCalledTimes(1);

  expect(container).toMatchSnapshot();
});
