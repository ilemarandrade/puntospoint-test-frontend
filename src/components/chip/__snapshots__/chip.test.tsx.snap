// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders and calls onClick 1`] = `
<div>
  <div
    aria-disabled="true"
    class="MuiButtonBase-root Mui-disabled MuiChip-root MuiChip-filled Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1ykktcv-MuiButtonBase-root-MuiChip-root"
    role="button"
    tabindex="-1"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
    >
      Test Label
    </span>
  </div>
</div>
`;

exports[`Chip renders outlined chip 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-outlined Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1lrxra9-MuiChip-root"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
    >
      Test Label
    </span>
  </div>
</div>
`;

exports[`Chip renders outlined chip with delete 1`] = `
<div>
  <div
    class="MuiButtonBase-root MuiChip-root MuiChip-outlined Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-outlinedDefault css-1gc4ec1-MuiButtonBase-root-MuiChip-root"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
    >
      Test Label
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconOutlinedColorDefault css-1umw9bq-MuiSvgIcon-root"
      data-testid="ClearRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7a.996.996 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41s1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0s.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4"
      />
    </svg>
  </div>
</div>
`;

exports[`Chip renders outlined chip with icon 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-outlined Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1lrxra9-MuiChip-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-1umw9bq-MuiSvgIcon-root"
      data-testid="CarCrashRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 1c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5m0 6c-.28 0-.5-.22-.5-.5v-3c0-.28.22-.5.5-.5s.5.22.5.5v3c0 .28-.22.5-.5.5m.5 1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5.22-.5.5-.5.5.22.5.5m1 11.5c.82 0 1.5-.67 1.5-1.5v-6.18c-1.05.51-2.16.69-3.09.68.06.16.09.33.09.5 0 .83-.67 1.5-1.5 1.5s-1.5-.67-1.5-1.5c0-.39.15-.74.39-1.01-1.63-.66-2.96-1.91-3.71-3.49H5.81l1.04-3H11c0-.69.1-1.37.29-2H6.5c-.66 0-1.21.42-1.42 1.01l-1.97 5.67c-.07.21-.11.43-.11.66v7.16c0 .83.67 1.5 1.5 1.5S6 19.33 6 18.5V18h12v.5c0 .83.68 1.5 1.5 1.5m-12-5c-.83 0-1.5-.67-1.5-1.5S6.67 12 7.5 12s1.5.67 1.5 1.5S8.33 15 7.5 15"
      />
    </svg>
    <span
      class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
    >
      Test Label
    </span>
  </div>
</div>
`;

exports[`Chip renders with delete icon and calls onDelete 1`] = `
<div>
  <div
    class="MuiButtonBase-root MuiChip-root MuiChip-filled Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-16yb58i-MuiButtonBase-root-MuiChip-root"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
    >
      Test Label
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-1umw9bq-MuiSvgIcon-root"
      data-testid="ClearRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7a.996.996 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41s1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0s.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4"
      />
    </svg>
  </div>
</div>
`;

exports[`Chip renders with icon 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-filled Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-5c51lm-MuiChip-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-1umw9bq-MuiSvgIcon-root"
      data-testid="CarCrashRoundedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 1c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5m0 6c-.28 0-.5-.22-.5-.5v-3c0-.28.22-.5.5-.5s.5.22.5.5v3c0 .28-.22.5-.5.5m.5 1.5c0 .28-.22.5-.5.5s-.5-.22-.5-.5.22-.5.5-.5.5.22.5.5m1 11.5c.82 0 1.5-.67 1.5-1.5v-6.18c-1.05.51-2.16.69-3.09.68.06.16.09.33.09.5 0 .83-.67 1.5-1.5 1.5s-1.5-.67-1.5-1.5c0-.39.15-.74.39-1.01-1.63-.66-2.96-1.91-3.71-3.49H5.81l1.04-3H11c0-.69.1-1.37.29-2H6.5c-.66 0-1.21.42-1.42 1.01l-1.97 5.67c-.07.21-.11.43-.11.66v7.16c0 .83.67 1.5 1.5 1.5S6 19.33 6 18.5V18h12v.5c0 .83.68 1.5 1.5 1.5m-12-5c-.83 0-1.5-.67-1.5-1.5S6.67 12 7.5 12s1.5.67 1.5 1.5S8.33 15 7.5 15"
      />
    </svg>
    <span
      class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
    >
      Test Label
    </span>
  </div>
</div>
`;

exports[`Chip renders with label 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-filled Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-5c51lm-MuiChip-root"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
    >
      Test Label
    </span>
  </div>
</div>
`;

exports[`renders outlined chip and calls onClick 1`] = `
<div>
  <div
    aria-disabled="true"
    class="MuiButtonBase-root Mui-disabled MuiChip-root MuiChip-outlined Mui-disabled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-soe3uv-MuiButtonBase-root-MuiChip-root"
    role="button"
    tabindex="-1"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
    >
      Test Label
    </span>
  </div>
</div>
`;
